test/integration/client/exec_test.go

```diff
@@ -407,6 +382,135 @@ func TestExecPlugin(t *testing.T) {
 	}
 }
 
+type oldNew struct {
+	old, new interface{}
+}
+
+type informerSpy struct {
+	adds    []interface{}
+	updates []oldNew
+	deletes []interface{}
+}
+
+func (es *informerSpy) OnAdd(obj interface{}) { es.adds = append(es.adds, obj) }
+
+func (es *informerSpy) OnUpdate(old, new interface{}) {
+	es.updates = append(es.updates, oldNew{old: old, new: new})
+}
+
+func (es *informerSpy) OnDelete(obj interface{}) { es.deletes = append(es.deletes, obj) }
+
+func TestExecPluginViaInformer(t *testing.T) {
```

Comment:
While I think the existing `2` tests are fine as a starting point, we really need to test the informer over time - i.e., if the creds under it rotate, does it keep working (and if the creds stop working for some time, and then start working again, does it recover, ...).

Reply:
Totes agreed. That is the next step. Figured I would start small to follow a "simple PR first" principle.